charts/drone-vault-secrets/values.yaml

```diff
@@ -0,0 +1,103 @@
+##
+imagePullSecrets: []
+# - name: "image-pull-secret"
+
+## For small or experimental deployments of the Kubernetes runner, 1 replica will suffice.
+## For production cases, 2-3 are recommended. This does not grant additional parallelism,
+## but does ensure that upgrades, config changes, and disruptions are handled more gracefully.
+replicaCount: 1
+
+nameOverride: ""
+fullnameOverride: ""
+
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: true
+  # Annotations to add to the service account
+  annotations: {}
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name: ""
+
+podAnnotations: {}
+
+podSecurityContext: {}
+  # fsGroup: 2000
+
+securityContext: {}
+  # capabilities:
+  #   drop:
+  #   - ALL
+  # readOnlyRootFilesystem: true
+  # runAsNonRoot: true
+  # runAsUser: 1000
+
+service:
+  type: ClusterIP
+  port: 3000
+
+resources: {}
+  # limits:
+  #   cpu: 100m
+  #   memory: 128Mi
+  # requests:
+  #   cpu: 100m
+  #   memory: 128Mi
+
+nodeSelector: {}
+
+tolerations: []
+
+affinity: {}
+
+rbac:
+  ## If true, create a Role + Rolebinding in the secretNamespace that allows the extension to
+  ## fetch Kubernetes Secrets.
+  ##
+  enabled: true
+  ## The namespace that the extension is allowed to fetch secrets from. Unless
+  ## rbac.restrictToSecrets is set below, the extension will be able to pull all secrets in
+  ## the namespace specified here.
+  ##
+  secretNamespace: default
+  ## Optionally restrict secret pulls to a subset of secrets in the secret namespace. This is
+  ## particularly useful if your secrets extension is fetching secrets from a namespace that other
+  ## Drone components have been deployed to.
+  restrictToSecrets: []
+
+## If you'd like to provide your own Kubernetes Secret object instead of passing your values
+## in un-encrypted, pass in the name of a created + populated Secret in the same Namespace
+## as the pod. All secrets within this configmap will be mounted as environment variables,
+## with each key/value mapping to a corresponding environment variable on the pod.
+##
+extraSecretNamesForEnvFrom: []
+# - my-drone-secrets
+
+## The keys within the "env" map are mounted as environment variables on the secrets extension pod.
+##
+env:
+  ## REQUIRED: Shared secret value for comms between the Kubernetes runner and this secrets plugin.
+  ## Must match the value set in the runner's env.DRONE_SECRET_PLUGIN_TOKEN.
+  ## Ref: https://kube-runner.docs.drone.io/installation/reference/drone-secret-plugin-token/
+  ## This is commented out in order to leave you the ability to set the
+  ## key via a separately provisioned secret (see existingSecretName above).
+  ##
+  # DRONE_SECRET:
+  ##
+  ##
+  ## Vault related parameter
+  ## For more information about Vault extension parameter, you can take a look here:
+  ## https://github.com/drone/drone-vault
```

Comment:
We need some examples here of using different `VAULT_AUTH_TYPE`s including these and other VAULT vars:
- token
    - `VAULT_TOKEN`
- kubernetes
    - `VAULT_KUBERNETES_ROLE`
- approle
    - `VAULT_APPROLE_ID`
    - `VAULT_APPROLE_SECRET`

Like DRONE_SECRET, this will be set  via `extraSecretNamesForEnvFrom`, but we need the examples to clarify how to use the role.

Reply:
```suggestion
  ## https://github.com/drone/drone-vault
  ## https://www.vaultproject.io/docs/commands#environment-variables
  ##
  # VAULT_ADDR: vault-example.local
  ##
  ## Examples of various auth types (use extraSecretNamesForEnvFrom above for these):
  ## (example values from https://learn.hashicorp.com/tutorials/vault/getting-started-apis)
  ##
  # VAULT_AUTH_TYPE: kubernetes
  # VAULT_KUBERNETES_ROLE: drone
  ##
  # VAULT_AUTH_TYPE: approle
  # VAULT_APPROLE_ID: 3c301960-8a02-d776-f025-c3443d513a18
  # VAULT_APPROLE_SECRET: 22d1e0d6-a70b-f91f-f918-a0ee8902666b
  ##
  # VAULT_AUTH_TYPE: token
  # VAULT_TOKEN: s.Ga5jyNq6kNfRMVQk2LY1j9iu
```